app/components/UI/Bridge/components/TransactionDetails/BridgeStepDescription.tsx

```
@@ -0,0 +1,203 @@
+  NETWORK_TO_SHORT_NETWORK_NAME_MAP,
+} from '../../../../../constants/bridge';
+import { Box } from '../../../Box/Box';
+import { AlignItems, FlexDirection } from '../../../Box/box.types';
+import Text, {
+  TextColor,
+  TextVariant,
+} from '../../../../../component-library/components/Texts/Text';
+import { strings } from '../../../../../../locales/i18n';
+import { StyleSheet } from 'react-native';
+import { Step } from '@metamask/bridge-controller';
+
+/**
+ * bridge actions will have step.srcChainId !== step.destChainId
+ * We cannot infer the status of the bridge action since 2 different chains are involved
+ * The best we can do is the bridgeHistoryItem.estimatedProcessingTimeInSeconds
+ *
+ * @param stepStatus - The status of the step
+ * @param step - The step to be rendered
+ * @param networkConfigurationsByChainId - The network configurations by chain id
+ */
+const getBridgeActionText = (
+  stepStatus: StatusTypes | null,
+  step: Step,
+  networkConfigurationsByChainId: Record<`0x${string}`, NetworkConfiguration>,
+) => {
+  const hexDestChainId = step.destChainId
+    ? decimalToPrefixedHex(step.destChainId)
+    : undefined;
+  const destNetworkConfiguration = hexDestChainId
+    ? networkConfigurationsByChainId[hexDestChainId as Hex]
+    : undefined;
+
+  const destChainName =
+    NETWORK_TO_SHORT_NETWORK_NAME_MAP[
+      destNetworkConfiguration?.chainId as AllowedBridgeChainIds
+    ];
+
+  const destSymbol = step.destAsset?.symbol;
+
+  if (!destSymbol) {
+    return null;
+  }
+
+  return stepStatus === StatusTypes.COMPLETE
+    ? strings('bridge_transaction_details.bridge_step_action_bridge_complete', {
+        destSymbol,
+        destChainName,
+      })
+    : strings('bridge_transaction_details.bridge_step_action_bridge_pending', {
+        destSymbol,
+        destChainName,
+      });
+};
+
+const getBridgeActionStatus = (bridgeHistoryItem: BridgeHistoryItem) =>
+  bridgeHistoryItem.status ? bridgeHistoryItem.status.status : null;
```

Is null a valid scenario or does it represent an error condition? If it's the latter, let's also report an error